wiki/Bazel-Setup-Instructions-for-Mac.md

```diff
@@ -0,0 +1,122 @@
+## Bazel Setup for Mac including Apple silicon (M1/M2) chips
+
+## Instructions
+
+**The steps to install Bazel on Mac are:**
+1. Setup Rosetta Terminal
+2. Install Bazel 
+3. Install OpenJDK 8
+4. Install Python 2 and make sure it is active in your environment
+5. Set up the ANDROID_HOME environment variable
+6. Set Bazel, Python 2, ANDROID_HOME paths permanently in your terminal
+7. Prepare the build environment
+8. Verify that the build is working
+
+### 1. Setup Rosetta Terminal
+
+- In the Finder app on your Mac, locate the Applications folder from the favorites sidebar.
+- Right-click on your Terminal app and create a duplicate Terminal (and rename it accordingly, say **Terminal Rosetta**, to avoid confusion).
+- On the newly created Terminal Rosetta icon, right-click and select "Get info", and under “General”, check the option "Open using Rosetta".
+
+**Note: Always use the Rosetta terminal for Bazel setup and running `bash setup.sh` or any Bazel build-related commands.**
+
+### 2. Install Bazel
+
+1. Install Bazel following the instructions [here](https://docs.bazel.build/versions/4.0.0/install-os-x.html#install-with-installer-mac-os-x). Make sure that you follow the instructions for installing a specific version (Oppia Android requires 4.0.0 and won't build on other versions). 
+
+2. That’s it, now Bazel is setup permanently in your terminal, you can again check it by running the command:
+   ```
+   bazel --version
+   ```
+   - **Expected Output**
+   ```
+   bazel 4.0.0
+   ```
+
+**Note: You must set the path for `bazel 4.0.0` before running bazel build for oppia-android, otherwise you will get an error.**
+
+### 3. Install OpenJDK 8
+
+Oppia Android also requires OpenJDK 8.
+Follow the instructions [here](https://installvirtual.com/install-openjdk-8-on-mac-using-brew-adoptopenjdk/) to install OpenJDK 8. 
+Note that this requires the installation of brew as a pre-requisite, which can be done by following the instructions [here](https://mac.install.guide/homebrew/index.html). You can then set up your `$JAVA_HOME` environment variable using these [instructions](https://stackoverflow.com/a/75167958/11396524).
+
+
+### 4. Installing Python 2
+
+```
```

Comment:
Write something before this, e.g. "To do XYZ, run the following commands:"

Reply:
Done.